src/main/java/org/damap/base/rest/invenio_damap/InvenioDamapResourceMapper.java

```diff
@@ -0,0 +1,172 @@
+package org.damap.base.rest.invenio_damap;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import lombok.experimental.UtilityClass;
+import lombok.extern.jbosslog.JBossLog;
+import org.damap.base.enums.EAccessRight;
+import org.damap.base.enums.EDataAccessType;
+import org.damap.base.enums.EDataKind;
+import org.damap.base.enums.EDataSource;
+import org.damap.base.enums.EDataType;
+import org.damap.base.enums.EIdentifierType;
+import org.damap.base.enums.ELicense;
+import org.damap.base.rest.dmp.domain.DatasetDO;
+import org.damap.base.rest.dmp.domain.DmpDO;
+import org.damap.base.rest.dmp.domain.ExternalStorageDO;
+import org.damap.base.rest.dmp.domain.IdentifierDO;
+import org.damap.base.rest.madmp.dto.Dataset;
+import org.damap.base.rest.madmp.dto.Host;
+
+@JBossLog
+@UtilityClass
+public class InvenioDamapResourceMapper {
+  public DatasetDO mapMaDMPDatasetToDatasetDO(
+      Dataset madmpDataset, DatasetDO datasetDO, DmpDO dmpDO) {
+
+    // Disclaimer: This is by no means complete. Not all fields of the
+    // Dataset or DMP are set. Null value checks should also be performed.
+    var datasetId = madmpDataset.getDatasetId();
+    if (datasetId != null) {
+      IdentifierDO newId = new IdentifierDO();
+      newId.setIdentifier(datasetId.getIdentifier());
+      newId.setType(EIdentifierType.valueOf(datasetId.getType().name().toUpperCase()));
+      datasetDO.setDatasetId(newId);
+    }
+
+    datasetDO.setReferenceHash((new Date()).toString());
+    datasetDO.setDateOfDeletion(null);
+    datasetDO.setDelete(false);
+    datasetDO.setDeletionPerson(null);
+    datasetDO.setDescription(madmpDataset.getDescription());
+    datasetDO.setLegalRestrictions(null);
+    datasetDO.setLicense(null);
+    datasetDO.setSize(0L);
+
+    mapDistribution(madmpDataset, datasetDO, dmpDO);
+
+    datasetDO.setOtherProjectMembersAccess(EAccessRight.READ);
+
+    Boolean personalData = true;
+    madmpDataset.setPersonalData(
+        Objects.requireNonNullElse(madmpDataset.getPersonalData(), Dataset.PersonalData.UNKNOWN));
+    switch (madmpDataset.getPersonalData()) {
```

Comment:
It is not necessary to initialize the personalData variable.
It is also unnecessary to set personalData in madmpDataset - `Objects.requireNonNullElse(madmpDataset.getPersonalData(), Dataset.PersonalData.UNKNOWN))` can be used in the switch statement instead of `madmpDataset.getPersonalData()`.

the same is also true for the sensitiveData part.